File: sdkwork-backend-sdk-csharp/Api/ImageApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class ImageApi
    {
        private readonly HttpClient _client;

        public ImageApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update an existing image
        /// </summary>
        public async Task<PlusApiResultPlusImageVO?> UpdateAsync(PlusImageForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusImageVO>(ApiPaths.BackendPath("/image"), body);
        }

        /// <summary>
        /// Create a new image
        /// </summary>
        public async Task<PlusApiResultPlusImageVO?> CreateAsync(PlusImageForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusImageVO>(ApiPaths.BackendPath("/image"), body);
        }

        /// <summary>
        /// Get images by page
        /// </summary>
        public async Task<PlusApiResultPagePlusImageVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusImageVO>(ApiPaths.BackendPath("/image/list"), body, query);
        }

        /// <summary>
        /// Get all images
        /// </summary>
        public async Task<PlusApiResultListPlusImageVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusImageVO>(ApiPaths.BackendPath("/image/list/all"), body);
        }

        /// <summary>
        /// Get an image by ID
        /// </summary>
        public async Task<PlusApiResultPlusImageVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusImageVO>(ApiPaths.BackendPath($"/image/{id}"));
        }

        /// <summary>
        /// Delete an image
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/image/{id}"));
        }
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/ImageApi.swift
import Foundation

public class ImageApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing image
    public func update(body: PlusImageForm) async throws -> PlusApiResultPlusImageVO? {
        let response = try await client.put(ApiPaths.backendPath("/image"), body: body)
        return response as? PlusApiResultPlusImageVO
    }

    /// Create a new image
    public func create(body: PlusImageForm) async throws -> PlusApiResultPlusImageVO? {
        let response = try await client.post(ApiPaths.backendPath("/image"), body: body)
        return response as? PlusApiResultPlusImageVO
    }

    /// Get images by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusImageVO? {
        let response = try await client.post(ApiPaths.backendPath("/image/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusImageVO
    }

    /// Get all images
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusImageVO? {
        let response = try await client.post(ApiPaths.backendPath("/image/list/all"), body: body)
        return response as? PlusApiResultListPlusImageVO
    }

    /// Get an image by ID
    public func getById(id: String) async throws -> PlusApiResultPlusImageVO? {
        let response = try await client.get(ApiPaths.backendPath("/image/\(id)"))
        return response as? PlusApiResultPlusImageVO
    }

    /// Delete an image
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/image/\(id)"))
        return response as? PlusApiResultBoolean
    }
}
